### ISSUE TITLE
Are CSV paths for readinessProbe and livenessProbe reversed?

### ISSUE THREAD

**Issue by jwalcorn** [operator-framework/operator-sdk#4601]
## Bug Report

Looking at the generated CSV yaml, under spec.install.spec.deployments, there are two containers: `kube-rbac-proxy`
and `manager`.  The latter has a livenessProbe and a readinessProbe defined.  But the paths look swapped: the `/readyz` path is on the livenessProbe, not on the readinessProbe as you'd expect from a path named "readyz".  The path on the readinessProbe is `/healthz`.  I'm wondering if a copy/paste typo got made here, swapping the two paths.

#### What did you do?

Generated a helm-based operator and looked at the yaml for the CSV.

#### What did you expect to see?

```
                livenessProbe:
                  httpGet:
                    path: /healthz
                    port: 8081
                  initialDelaySeconds: 15
                  periodSeconds: 20
                readinessProbe:
                  httpGet:
                    path: /readyz
                    port: 8081
                  initialDelaySeconds: 5
                  periodSeconds: 10
```

#### What did you see instead? Under which circumstances?

```
                livenessProbe:
                  httpGet:
                    path: /readyz
                    port: 8081
                  initialDelaySeconds: 15
                  periodSeconds: 20
                readinessProbe:
                  httpGet:
                    path: /healthz
                    port: 8081
                  initialDelaySeconds: 5
                  periodSeconds: 10
```

#### Environment

**Operator type:**

/language helm

**Kubernetes cluster type:**

OpenShift

`$ operator-sdk version`

v1.4.0

`$ go version` (if language is Go)

`$ kubectl version`

v1.17.2

#### Possible Solution

Swap the paths for the readinessProbe and the livenessProbe

#### Additional context

If the values are somehow correct, perhaps at least generate a comment explaining the confusion

**Comment by jwalcorn** [comment]
BTW, I'm getting an image pull error on that `manager` container.  The `image` field is set to just `controller:latest`, without a registry or org specified.  When I install my operator, I see the following in the events for the pod for my operator:
```
Failed to pull image "controller:latest": rpc error: code = Unknown desc = Error reading manifest latest in docker.io/library/controller: errors: denied: requested access to the resource is denied unauthorized: authentication required
```
Do we have to have our users create an ImagePullSecret in order to get our operators working?  If so, is there a way I can get my install plan to make that for the user?  Thanks.

![image](https://user-images.githubusercontent.com/28963246/109922650-4ed3de80-7c83-11eb-8dbc-f1d841fcbb6c.png)

My CSV is here: https://github.com/IBMStockTrader/stocktrader-operator/blob/master/bundle/manifests/stocktrader-operator.clusterserviceversion.yaml

**Comment by estroz** [comment]
@jwalcorn the endpoint mixup was fixed in #4546. The v1.5 migration guide, when released, will discuss swapping them.

The intent is to run `make bundle IMG=<registry>/<repo>/your-operator:vX.Y.Z` so kustomize can substitute the default image name `controller:latest` for your remote image. If you still have questions about this, open another issue. Thanks!

**Comment by jwalcorn** [comment]
Thanks, glad to hear the swapped endpoints will be fixed in 1.5.  And yes, I had separately figured out what value needed to be in that image field for the manager container, by looking at the sample at https://github.com/operator-framework/operator-sdk/tree/master/testdata/helm/memcached-operator.
